### ISSUE TITLE
Test case failing across many different test files

### ISSUE THREAD

**Issue by jowhee3011** [nus-cs2103-AY2425S1/forum#494]
Hello, I've made some minor changes to the add command feature, however, it has been causing me gradle build errors as I am failing the given test cases, even those that aren't related to add command.

Most of the errors prompt me the following.
![image](https://github.com/user-attachments/assets/73a729c6-0ad8-40a6-b302-3d1fe28a8428)

Is there any way I can go about this? I am rather sure my code changes do not affect the test cases.

PR: https://github.com/AY2425S1-CS2103T-T11-3/tp/pull/42

**Comment by damithc** [comment]
@jowhee3011 A general approach to locate the cause of a problem is to go back to the last known 'correct' state, and incrementally add the changes while running tests after each change. If you add only a few lines at a time, you'll be able to locate which few lines cause the tests to start failing. For example, 

1. Go back to master branch. Start a new branch. Run tests to ensure they pass.
1. Repeat a small part of the changes in this branch. Run tests. If they pass, commit those changes.
1. Keep doing the above step until tests start to fail. When they do, the uncommitted changes are the lines you need to look into.

If using that approach, you need to be running tests locally, not relying on the PR CI runs (the latter will take too long).

**Comment by Domokunx** [comment]
Good morning!

I did a quick skim of the test cases you failed and I shall share some insights into what I think is causing some of the errors!

For your Phone.java, you modified the REGEX_VALIDATION to only allow 8-digit phone numbers (reasonable).
- You forgot to update the TypicalPersons.java phone numbers, with some of the Persons having 7 digit phone numbers. (See below)
![image](https://github.com/user-attachments/assets/6591aa21-b2c9-479a-aada-d64a949451ad)

Just to confirm I forked your branch, updated TypicalPerson.java and ran the tests. I went from failed 72 tests to 6 tests. Hope this helps despite being a little late.
